exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServer.java

```diff
@@ -82,7 +83,13 @@ public static PrometheusHttpServerBuilder builder() {
     // sequentially.
     if (memoryMode == MemoryMode.REUSABLE_DATA) {
       executor =
-          Executors.newSingleThreadExecutor(new DaemonThreadFactory("prometheus-http-server"));
```

Comment:
`newSingleThreadExecutor` returns a delegating wrapped `ThreadPoolExecutor`. One of our tests introspects on the resolved executor and ensures that when memory mode is reuseable, a single thread executor is used. The test is not able to make the required assertions based on the delegating wrapped version returned by `Executors.newSingleThreadExecutor`.

Reply:
The way you described it makes it sound like a shortcoming in the test/assertions...which is a bummer that the implementation had to change to accommodate it. It seems fine, but I can't fully reason about the impact of having a non-finalizable executorservice. I'm overthinking it.